api/src/main/java/io/minio/GetPresignedObjectUrlArgs.java

```diff
@@ -68,10 +70,14 @@ public Builder expiry(int expiry) {
     }
 
     public Builder expiry(int duration, TimeUnit unit) {
-      int expiry = (int) TimeUnit.SECONDS.convert(duration, unit);
-      validateExpiry(expiry);
-      operations.add(args -> args.expiry = expiry);
-      return this;
+      return expiry((int) unit.toSeconds(duration));
+    }
+
+    @Override
+    protected void validate(GetPresignedObjectUrlArgs args) {
+      super.validate(args);
+      validateMethod(args.method);
+      validateExpiry(args.expiry);
```

Comment:
`validateExpiry()` is not needed here because `args.expiry` has default value and no way to set invalid value to this.